### PR TITLE
Add e2e benchmarks for aggregation jobs

### PR DESCRIPTION
**Pull request by mendess**
- Move report generator from dapf to daphne
- Move vdaf measurment gen method from dapf to daphne
- Add benchmark around aggregation jobs
- Make the test implementation of ReportInitializer as fast as the daphne-server one

### CODE REVIEW

File: crates/daphne/benches/aggregation.rs
Comment (cjpatton) [inline crates/daphne/benches/aggregation.rs:1]: This doesn't compile.

**Review by cjpatton — CHANGES_REQUESTED** [review]

File: crates/daphne/benches/aggregation.rs
Comment (cjpatton) [inline crates/daphne/benches/aggregation.rs:1]: Hmm, something strange might be happening with the benchmarks. When I ran `cargo bench`, I sew things taking nanoseconds, where in fact we would expect these operations to take milliseconds:

```
prio3-sum-vec-multiproof-100/1
                        time:   [779.34 ns 781.49 ns 783.68 ns]
                        thrpt:  [1.2760 Melem/s 1.2796 Melem/s 1.2831 Melem/s]
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild

prio3-sum-vec-multiproof-1000/1
                        time:   [782.79 ns 784.91 ns 787.35 ns]
                        thrpt:  [1.2701 Melem/s 1.2740 Melem/s 1.2775 Melem/s]
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild

prio3-sum-vec-multiproof-10000/1
                        time:   [778.33 ns 779.59 ns 780.84 ns]
                        thrpt:  [1.2807 Melem/s 1.2827 Melem/s 1.2848 Melem/s]
Found 3 outliers among 100 measurements (3.00%)
  3 (3.00%) high mild

prio3-sum-vec-multiproof-100000/1
                        time:   [779.80 ns 781.73 ns 783.88 ns]
                        thrpt:  [1.2757 Melem/s 1.2792 Melem/s 1.2824 Melem/s]
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild

prio3-sum-vec-multiproof-100/10
                        time:   [769.24 ns 770.71 ns 772.43 ns]
                        thrpt:  [12.946 Melem/s 12.975 Melem/s 13.000 Melem/s]
Found 9 outliers among 100 measurements (9.00%)
  6 (6.00%) high mild
  3 (3.00%) high severe

prio3-sum-vec-multiproof-1000/10
                        time:   [771.83 ns 772.88 ns 774.08 ns]
                        thrpt:  [12.919 Melem/s 12.939 Melem/s 12.956 Melem/s]
Found 26 outliers among 100 measurements (26.00%)
  3 (3.00%) low severe
  13 (13.00%) low mild
  4 (4.00%) high mild
  6 (6.00%) high severe

prio3-sum-vec-multiproof-10000/10
                        time:   [774.47 ns 775.38 ns 776.36 ns]
                        thrpt:  [12.881 Melem/s 12.897 Melem/s 12.912 Melem/s]
Found 9 outliers among 100 measurements (9.00%)
  2 (2.00%) low mild
  2 (2.00%) high mild
  5 (5.00%) high severe
```

Note that we noticed this earlier as well: https://github.com/cloudflare/daphne/issues/368

File: crates/daphne/src/vdaf/mastic.rs [hunk 1]
```diff
@@ -54,11 +56,11 @@ pub enum MasticWeight {
     Bool(bool),
 }
 
-pub(crate) fn mastic_shard(
+pub(crate) fn mastic_shard<const NUM_AGGREGATORS: usize>(
```
Comment (cjpatton) [inline crates/daphne/src/vdaf/mastic.rs:59]: Why do we need this? NUM_AGGREGATORS will always be 2 in DAP, so i think you could just hard-code 2 in the result below.

**Review by cjpatton — APPROVED** [review]

File: crates/daphne/src/vdaf/mastic.rs (hunk 1 above)
Comment (mendess) [reply crates/daphne/src/vdaf/mastic.rs:59]: maybe I was a bit too fancy :sweat_smile:

**Pull request merged**